.github/workflows/test.yml

```diff
@@ -59,3 +59,57 @@ jobs:
           chmod -R +x bin
           make completions-generation
           hack/tree_status.sh
+
+  mocks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.15'
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: go-mocks-${{ hashFiles('**/go.sum') }}
+          restore-keys: go-mocks-
+      - run: scripts/github-actions-packages
+      - run: |
+          make mockgen -j6
+          hack/tree_status.sh
+
+  unit:
+    needs: mocks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.15'
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: go-unit-${{ hashFiles('**/go.sum') }}
+          restore-keys: go-unit-
+      - run: scripts/github-actions-packages
+      - run: make testunit
+      - uses: actions/upload-artifact@v2
+        with:
+          name: unit
+          path: |
+            build/coverage
+            build/junit
+
+  coverage:
+    needs: unit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
+        with:
+          name: unit
+          path: build
```

Comment:
What is the reason for having a separate `make coverage` job? It seems it could just be the last step of `unit` job, which would simplify things a lot.

Reply:
IOW
```suggestion
```

Reply:
Yeah I agree, merged both steps into a single one :+1: